sql/src/main/java/io/crate/operation/scalar/conditional/IfFunction.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.conditional;
+
+import com.google.common.base.Preconditions;
+import io.crate.analyze.symbol.Function;
+import io.crate.metadata.*;
+import io.crate.operation.Input;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.types.BooleanType;
+import io.crate.types.DataType;
+
+import java.util.List;
+import java.util.Locale;
+
+public class IfFunction extends Scalar<Object, Object> {
```

Comment:
Consider adding javadocs with a simple synopsis.
E.g  if (condition, result, defaultValue) or something.
For me it was unclear what the args.length == 3 case is.